packages/mgt-chat/src/components/ChatList/ChatList.tsx

```diff
@@ -88,6 +89,22 @@ export const ChatList = (
     setMenuItems(menuItems);
   }, []);
 
+  // Store last read time in cache so that when the user comes back to the chat list,
+  // we know what messages they are likely to have not read. This is not perfect because
+  // the user could have read messages in another client (for instance, the Teams client).
+  useEffect(() => {
+    const timer = setInterval(() => {
+      if (selectedItem) {
+        log(`caching the last-read timestamp of now to chat ID '${selectedItem}'...`);
+        cache.cacheLastReadTime(selectedItem, new Date());
+      }
+    }, lastReadTimeInterval);
```

Comment:
because `lastReadTimeInterval` is configurable, would we need to write some guard here to prevent null, undefined, some other negative values?

Reply:
That's a fair question. Generally I would only put guards like that when the values could be supplied by an end-user not a developer. If the developer uses null or undefined, it will use the default so no problem there. If they used a negative value, I am not sure what happens, but it seems like a developer couldn't be doing that by mistake. Are there any examples in MGT where the PG is doing validation on input values other than Typescript constraints?

Reply:
There are no guards that I can see. In my PR I did put a guard for the page count because I do not want it to be an invalid value since I am passing that into my HTTP request.